ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -74,4 +77,19 @@ public void shouldFailWhenSyntaxIsInvalid() throws Exception {
     ksqlEngine.createQueries("blah;");
   }
 
+  @Test
+  public void shouldEnforceTopicExistanceCorrectly() throws Exception {
+    topicClient.createTopic("s1_topic", 1, (short) 1);
+    StringBuilder runScriptContent =
+        new StringBuilder("CREATE STREAM S1 (COL1 BIGINT, COL2 VARCHAR) "
+                          + "WITH  (KAFKA_TOPIC = 's1_topic', VALUE_FORMAT = 'JSON');\n");
+    runScriptContent.append("CREATE TABLE T1 AS SELECT COL1, count(*) FROM "
+                            + "S1 GROUP BY COL1;\n");
+    runScriptContent.append("CREATE STREAM S2 (C1 BIGINT, C2 BIGINT) "
```

Comment:
We probably also want to add a test to check that we fail in the case where the topic really doesn't exist. I can see that we drop the topic existence check during parsing. But I am not sure where we actually add that check back. 

In particular, if we have a `CREATE STREAM` DDL statement without an underlying topic, will we fail with this patch? I am not sure anymore

Reply:
I guess the check for existence happens when we actually execute a DDL statement, since `CommandFactories` is instantiated with `ensureTopicExistence == true`. But it would still be good to add a check for this.

Reply:
Added test to ensure we check the topic existence when it is needed.